tools/bazel_java_integration_test.bzl

```diff
@@ -97,15 +120,23 @@ def bazel_java_integration_test(
   else:
     runtime_deps = runtime_deps + add_deps
   for version in versions:
+    prop_rule = "%s/config%s" % (name, version)
+    bazel_java_integration_test_properties_(
+        name = prop_rule,
+        bazel_version = version,
+        external_deps = external_deps,
+    )
+
+    cur_data = data + external_deps + [prop_rule + ".properties"]
     if add_bazel_data_dependency:
-      data = data + [
-          "@build_bazel_bazel_%s//:bazel" % version.replace(".", "_")
-      ]
+      cur_data += ["@build_bazel_bazel_%s//:bazel" % version.replace(".", "_")]
     native.java_test(
         name = "%s/bazel%s" % (name, version),
-        jvm_flags = ["-Dbazel.version=" + version] + jvm_flags,
+        jvm_flags = [
+            "-Dbazel.configuration=$(location %s.properties)" % prop_rule
```

Comment:
@damienmg do you remember if this needs to be $execpath or something? I have a recollection that $location is for build time and we need runtime

Reply:
@ittaiz: this needs to be the runtime location, not the execpath. IIRC location give the runtime location so it should be good.

Reply:
Sorry, I meant the runtime location. I think location gives the runtime location for native rules. Not for skylark.

Reply:
From the [documentation](https://docs.bazel.build/versions/master/be/make-variables.html#location): _The expanded paths are relative to the runfiles directory of the \*\_test or \*\_binary rule._

So we are fine here as we are in a java_test.

(and also I meant *runfiles* not *runtime*)